agent/src/main/java/org/openremote/agent/protocol/modbus/ModbusAgent.java

```diff
@@ -19,31 +19,93 @@
  */
 package org.openremote.agent.protocol.modbus;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
 import jakarta.persistence.Entity;
 import org.openremote.model.asset.agent.Agent;
-import org.openremote.model.attribute.MetaItem;
-import org.openremote.model.value.AttributeDescriptor;
-import org.openremote.model.value.MetaItemType;
-import org.openremote.model.value.ValueConstraint;
-import org.openremote.model.value.ValueType;
+import org.openremote.model.value.ValueDescriptor;
+
+import java.util.HashMap;
 
 @Entity
 public abstract class ModbusAgent<T extends ModbusAgent<T, U>, U extends AbstractModbusProtocol<U, T>> extends Agent<T, U, ModbusAgentLink> {
 
-    public static final AttributeDescriptor<Integer> UNIT_ID = new AttributeDescriptor<>("unitId", ValueType.INTEGER, new MetaItem<>(MetaItemType.CONSTRAINTS, ValueConstraint.constraints(new ValueConstraint.Min(1), new ValueConstraint.Max(255))));
+    public enum EndianFormat {
+        BIG_ENDIAN,              // ABCD - Big byte order, Big word order
+        LITTLE_ENDIAN,           // DCBA - Little byte order, Little word order
+        BIG_ENDIAN_BYTE_SWAP,    // BADC - Big byte order, Little word order
+        LITTLE_ENDIAN_BYTE_SWAP; // CDAB - Little byte order, Big word order
+
+        @JsonValue
+        public String getJsonValue() {
+            return toString();
+        }
+    }
+
+    /**
+     * Configuration for a Modbus device (per unit ID).
+     * Contains all device-specific settings.
+     */
+    public static class ModbusDeviceConfig implements java.io.Serializable {
+        private EndianFormat endianFormat;
+        private String illegalRegisters;
+        private Integer maxRegisterLength;
+
+        @JsonCreator
+        public ModbusDeviceConfig(
+                @JsonProperty("endianFormat") EndianFormat endianFormat,
+                @JsonProperty("illegalRegisters") String illegalRegisters,
+                @JsonProperty("maxRegisterLength") Integer maxRegisterLength) {
```

Comment:
`@JsonProperty` annotations not needed as we compile code with method arg names included